accepted/global-json-updates.md

```diff
@@ -0,0 +1,315 @@
+# Proposal for global.json updates for preview and limited roll-forward
+
+---
+**NOTE:** 
+
+SDK version numbers do not use semantic versioning. [Docs has an article on SDK versioning](https://docs.microsoft.com/en-us/dotnet/core/versions/).
+
+---
+
+`global.json` allows users to pin the version of the SDK they are using. This was added to .NET Core in case we introduced an issue to allow roll-back per project until a fix could be provided. While our guidance remains to use it only for special circumstances, users now use it for numerous reasons.
+
+This document describes changes to `global.json` to provide more flexibility.
+
+## Definitions
+
+The term *SDK patch* is used to describe the last two positions of the SDK major/minor/feature/patch (n in x.y.znn).
+
+The term *feature band* is used to describe the SDK major/minor/feature/patch (x.y.z in x.y.znn).
+
+Given x.y.znn
+
+* x is the major version
+* y is the minor version
+* z is the feature band
+* nn is the patch
+
+## Current behavior
+
+The SDK resolvers (in both the CLI and VS) search from the current location (project in VS) up the directory hierarchy until they encounter a `global.json` file. When the resolver finds one, _it stop looking_ and attempts to resolve the SDK version specified in that file. This happens even if this `global.json` does not specify an SDK version.
+
+When a `global.json` is encountered and it specifies an SDK version, the SDK resolvers attempt to locate that specific SDK patch version:
+
+1. When the resolver finds no `global.json` or finds one that does not specify SDK version, the latest SDK is used, regardless of whether it is preview or stable.
+2. When the resolver finds a `global.json`, the specific SDK patch version it specifies is used if it is found.
+3. When the resolver finds a `global.json`, and the SDK it specifies can't be found, the resolver looks for a close match:
+    * Prior to .NET Core 2.1
+      * The highest patch version for the SDK feature band is used _regardless of whether it is higher than what is specified_. For example, if `global.json` specifies 2.2.103 is specified and is not available, and 2.2.101 is the highest SDK on the machine, it will be used.
+    * In .NET Core 2.1 and higher
+      * The highest patch version for the SDK feature band is used _assuming it is higher than what is specified_. For example, if `global.json` specifies 2.2.103 is specified, and 2.2.101 is the highest SDK on the machine, an error will occur.
```

Comment:
If I'm reading this correct we get the "greater than or equal to SDK version" feature by just specifying the version up to the feature brand level? That seems to be how the document reads but at the same time the behavior I'm describing has always seemed to be an ask vs. something that was already there.

Reply:
The behavior today is exact version if it exists, else latest from feature band.

Reply:
I'm assuming @nguerrera comment resolves @jaredpar 's. This is not the same as greater or equal, which is a feature added later in this proposal.